transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -219,7 +238,7 @@ public void run() {
         return promise;
     }
 
-    private void doReleaseChannel(Channel channel, Promise<Void> promise) {
+    private void doReleaseChannel(final Channel channel, final Promise<Void> promise, final boolean offerHealthyOnly) {
```

Comment:
Is it necessary to have this as an argument? Seems like we are just passing in the member variable when ever this is called.